pkg/datagatherer/istio/istio.go

```diff
@@ -0,0 +1,156 @@
+// Package isito provides a datagatherer for Istio.
+package istio
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"gopkg.in/yaml.v2"
+	"strings"
+	"time"
+
+	"github.com/jetstack/preflight/pkg/datagatherer/k8s"
+	"istio.io/istio/galley/pkg/config/analysis/analyzers"
+	"istio.io/istio/galley/pkg/config/analysis/local"
+	"istio.io/istio/pkg/config/resource"
+	istioSchema "istio.io/istio/pkg/config/schema"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"github.com/jetstack/preflight/pkg/datagatherer"
+	dgerror "github.com/jetstack/preflight/pkg/datagatherer/error"
+)
+
+// Config is the configuration for the Istio DataGatherer.
+type Config struct {
+	// KubeConfigPath is the path to the kubeconfig file. If empty, will assume it runs in-cluster.
+	KubeConfigPath string `yaml:"kubeconfig"`
+	// IstioNamespace is the Namespace for Istio components, this is usually `istio-system`.
+	IstioNamespace string `yaml:"istio-namespace"`
+	// ExcludeNamespaces is a list of namespaces to exclude.
+	ExcludeNamespaces []string `yaml:"exclude-namespaces"`
+	// IncludeNamespaces is a list of namespaces to include.
+	IncludeNamespaces []string `yaml:"include-namespaces"`
+	// Resources is a list of GroupVersionResources to collect for Istio analysis.
+	Resources []schema.GroupVersionResource
```

Comment:
Does this also need a yaml struct tag to support loading of a custom list?

Reply:
(if this isn't trivially easy, then I'm fine to leave that as a TODO for now)

Reply:
I've added a `YAML` tag for `Resources`. The `schema.GroupVersionResource` struct doesn't have YAML tags but I don't think they are required as its fields only have simple one word names.

This is also covered by the test as it unmarshals a config string that specifies resources.